src/org/labkey/test/util/exp/SampleTypeAPIHelper.java

```diff
@@ -33,15 +36,72 @@
      */
     public static TestDataGenerator createEmptySampleType(String containerPath, SampleTypeDefinition def)
     {
+        return createEmptySampleType(containerPath, def, null);
+    }
+
+    static public TestDataGenerator createEmptySampleType(String containerPath, SampleTypeDefinition sampleTypeDefinition, String dateFormatString)
+    {
+        deleteDomain(new FieldDefinition.LookupInfo(containerPath, "samples", sampleTypeDefinition.getName()));
+
+        TestDataGenerator dgen;
         try
         {
-            return TestDataGenerator.createDomain(containerPath, def);
+            dgen = TestDataGenerator.createDomain(containerPath, sampleTypeDefinition);
         }
         catch (CommandException e)
         {
             throw new RuntimeException("Failed to create sample type.", e);
         }
 
+        if (null != dateFormatString)
+        {
+            dgen.addDataSupplier("DateColumn", () -> dgen.randomDateString(dateFormatString,
+                    DateUtils.addWeeks(new Date(), -39),
+                    new Date()));
+        }
+        return dgen;
+    }
+
+    static public SampleTypeDefinition sampleTypeDefinition(String sampleTypeName,
+                                                            List<FieldDefinition> fields,
+                                                            String parentAlias,
+                                                            String nameExpression)
+    {
+        SampleTypeDefinition def = new SampleTypeDefinition(sampleTypeName);
+        def.setFields(fields);
+        if (null != parentAlias)
+            def.addParentAlias(parentAlias);
+        if (null != nameExpression)
+            def.setNameExpression(nameExpression);
+
+        return def;
+    }
```

Comment:
Why not add this as an overload to the constructor of SampleTypeDefinition?

Reply:
new SampleTypeDefinition(name).setFields(fields) is easy enough to call from test code.
Setting a parentAlias is simple enough for test code to handle

as a compromise I added a sample nameExpression to the helper class so test code won't have to be responsible for knowing what a valid name expression looks like